addons/portal/models/portal_mixin.py

```diff
@@ -109,7 +110,8 @@ def _get_access_action(self, access_uid=None, force_website=False):
     def action_share(self):
         action = self.env["ir.actions.actions"]._for_xml_id("portal.portal_share_action")
         action['context'] = {'active_id': self.env.context['active_id'],
-                             'active_model': self.env.context['active_model']}
+                             'active_model': self.env.context['active_model'],
+                             **literal_eval(action['context'])}
```

Comment:
```suggestion
                             'dialog_size': 'medium'}
```
why not do it directly like this ? instead of adding the attribute to the xml and overwriting the context here.

Reply:
Because `portal_share_action` is also called from some other actions and adding context in  `portal_share_action` and getting it here would be better then writing same context twice('dialog_size': 'medium') :innocent:
What do you think? :slightly_smiling_face:

Reply:
ah okay I see, you can leave it as is. Thanks for the info :muscle:

Reply:
I don't think the `literal_eval` is needed, we could instead doing something like this:

```suggestion
                             'active_model': self.env.context['active_model'],
                             **action['context']}
```

What do you think? 🙂

Reply:
We need `literal_eval` because we get string type context in `action['context']` and doing ** on string will give traceback :slightly_smiling_face: 
Is there any drawback of using `literal_eval`? :thinking: